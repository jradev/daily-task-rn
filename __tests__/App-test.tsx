/**
 * @format
 */

import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {FlatList, Text} from 'react-native';

it('renders correctly', () => {
  const tasks = [
    {id: 'AAB', title: 'Create Mobile App'},
    {id: 'BBA', title: 'Update Mobile App'},
  ];
  const tree = renderer
    .create(
      <FlatList
        data={tasks ?? []}
        renderItem={({item}) => <Text>{item.title}</Text>}
        keyExtractor={item => `task_id_${item?.id}`}
        extraData={tasks}
      />,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
