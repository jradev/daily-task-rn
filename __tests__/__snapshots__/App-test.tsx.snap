// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  data={
    [
      {
        "id": "AAB",
        "title": "Create Mobile App",
      },
      {
        "id": "BBA",
        "title": "Update Mobile App",
      },
    ]
  }
  extraData={
    [
      {
        "id": "AAB",
        "title": "Create Mobile App",
      },
      {
        "id": "BBA",
        "title": "Update Mobile App",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={[]}
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text>
        Create Mobile App
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text>
        Update Mobile App
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
